src/main/groovy/nebula/plugin/publishing/verification/PublishVerificationPlugin.groovy

```diff
@@ -44,23 +45,25 @@ class PublishVerificationPlugin implements Plugin<Project> {
 
     @CompileDynamic
     private void setupPlugin(Project project, PublishVerificationExtension extension) {
-        createVerificationViolationsCollector(project)
+        Provider<VerificationViolationsCollectorService> serviceProvider = project.getGradle().getSharedServices().registerIfAbsent("verificationViolationsCollectorService", VerificationViolationsCollectorService.class, spec -> {
```

Comment:
We are moving away from having an extension that was holding violation information to a build service approach that can be used by all tasks in a config cache safe way